app/src/main/java/org/wikipedia/notifications/NotificationActivity.kt

```diff
@@ -0,0 +1,643 @@
+    }
+
+    private fun deleteItems(items: List<NotificationListItemContainer>, markUnread: Boolean) {
+        val notificationsPerWiki: MutableMap<WikiSite, MutableList<Notification>> = HashMap()
+        val selectionKey = if (items.size > 1) Random().nextLong() else null
+        for (item in items) {
+            val wiki = if (dbNameMap.containsKey(item.notification!!.wiki())) dbNameMap[item.notification!!.wiki()]!! else WikipediaApp.getInstance().wikiSite
+            if (!notificationsPerWiki.containsKey(wiki)) {
+                notificationsPerWiki[wiki] = ArrayList()
+            }
+            notificationsPerWiki[wiki]!!.add(item.notification!!)
+            if (markUnread && !displayArchived) {
+                notificationList.add(item.notification!!)
+            } else {
+                notificationList.remove(item.notification)
+                NotificationFunnel(WikipediaApp.getInstance(), item.notification).logMarkRead(selectionKey)
+            }
+        }
+        for (wiki in notificationsPerWiki.keys) {
+            if (markUnread) {
+                NotificationPollBroadcastReceiver.markRead(wiki, notificationsPerWiki[wiki]!!, true)
+            } else {
+                NotificationPollBroadcastReceiver.markRead(wiki, notificationsPerWiki[wiki]!!, false)
+                showDeleteItemsUndoSnackbar(items)
+            }
+        }
+        postprocessAndDisplay()
+    }
+
+    private fun showDeleteItemsUndoSnackbar(items: List<NotificationListItemContainer>) {
+        val snackbar = FeedbackUtil.makeSnackbar(this, resources.getQuantityString(R.plurals.notification_archive_message, items.size, items.size), FeedbackUtil.LENGTH_DEFAULT)
+        snackbar.setAction(R.string.notification_archive_undo) { deleteItems(items, true) }
+        snackbar.show()
+    }
+
+    private fun finishActionMode() {
+        actionMode?.finish()
+    }
+
+    private fun beginMultiSelect() {
+        if (SearchActionModeCallback.`is`(actionMode)) {
+            finishActionMode()
+        }
+        if (!MultiSelectActionModeCallback.isTagType(actionMode)) {
+            startSupportActionMode(multiSelectActionModeCallback)
+        }
+    }
+
+    private fun toggleSelectItem(container: NotificationListItemContainer) {
+        container.selected = !container.selected
+        val selectedCount = selectedItemCount
+        if (selectedCount == 0) {
+            finishActionMode()
+        } else if (actionMode != null) {
+            actionMode!!.title = resources.getQuantityString(R.plurals.multi_items_selected, selectedCount, selectedCount)
+        }
+        binding.notificationsRecyclerView.adapter?.notifyDataSetChanged()
+    }
+
+    private val selectedItemCount: Int
+        get() {
+            var selectedCount = 0
+            for (item in notificationContainerList) {
+                if (item.selected) {
+                    selectedCount++
+                }
+            }
+            return selectedCount
+        }
```

Comment:
Can this be simplified to the following? 

```
private val selectedItemCount get() = notificationContainerList.count { it.selected }
```

Reply:
correct!